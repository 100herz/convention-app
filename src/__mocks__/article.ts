import { Article } from '@models/article'

export const article: Article = {
  id: 18569,
  date_gmt: '2020-11-11T08:25:41',
  title: {
    rendered: '„New Normal“ bei Leonardo Hotels &#8211; mit neuen MICE-Formaten und Konzepten',
  },
  content: {
    rendered:
      '<p>Durch smarte Raumanpassungen, große Flexibilität bei Buchungen und Stornierungen sowie die Etablierung eines extern zertifizierten und weit über den Standard hinaus gehenden Sicherheits- und Hygienekonzeptes hat sich Leonardo Hotels umfassend auf das „New Normal“ und die geänderten Anforderungen ihrer MICE-Kunden an zukunftsfähigen Veranstaltungsformaten eingestellt. Neue Pausenkonzepte, digitalen Helfer, verlässliche Partner sowie virtuelle und hybride Veranstaltungsformate umfassen die Angebotsvielfalt der Leonardo Hotels.  Diese sind überzeugt, dass persönliche Meetings mit Sicherheit wieder Zukunft haben.</p>\n',
    protected: false,
  },
  excerpt: {
    rendered:
      '<p>Durch neue Sicherheits- und Hygienekonzeptes hat sich Leonardo Hotels umfassend auf das „New Normal“ und die geänderten Anforderungen an zukunftsfähigen Veranstaltungsformaten eingestellt.</p>\n',
    protected: false,
  },
  featured_media: 18571,
  categories: [6, 1],
  acf: {
    sponsored_by: 'XYZ',
    should_push: false,
    pinned: false,
    featured_slider: false,
    news_slider: true,
  },
  featured_image_thumb:
    'https://www.convention-net.de/wp-content/uploads/2020/11/iStock-1250329916-Personen_mit_Masken-w1500-h1300-300x300.jpg',
  featured_image_medium:
    'https://www.convention-net.de/wp-content/uploads/2020/11/iStock-1250329916-Personen_mit_Masken-w1500-h1300-600x400.jpg',
  categories_names: ['Hotels', 'News'],
}

export const articleWithoutFeatured: Article = {
  id: 18592,
  date_gmt: '2020-11-10T11:00:19',
  title: {
    rendered:
      'Events and Hospitality Industries Urged to Harness the Power of Digital Transformation to Thrive in 2021 and Beyond',
  },
  content: {
    rendered:
      '<p><a href="https://bit.ly/3iPkUhB">Cvent</a> CEO and Founder, Reggie Aggarwal, delivered an inspiring keynote to more than 7,000 event, marketing, and hospitality professionals at <a href="https://bit.ly/30Unj4A">Cvent CONNECT Europe</a> today. In his opening presentation, Aggarwal urged attendees of the virtual event to seize the digital transformation opportunity in front of them to drive the industry’s recovery in 2021 and beyond.</p>\n',
    protected: false,
  },
  excerpt: {
    rendered:
      '<p>Cvent CEO Reggie Aggarwal predicts “Golden Age of Events” during his keynote to more than 7,000 virtual attendees at Cvent CONNECT Europe </p>\n',
    protected: false,
  },
  featured_media: null,
  categories: [358, 217, 1],
  acf: {
    sponsored_by: '',
    should_push: false,
    pinned: false,
    featured_slider: false,
    news_slider: true,
  },
  featured_image_thumb: '',
  featured_image_medium: '',
  categories_names: ['Branche', 'Events', 'News'],
}

export const articleWithoutSponsored: Article = {
  id: 18543,
  date_gmt: '2020-11-09T09:46:40',
  title: {
    rendered: 'Sicheres Business und Inspiration für morgen: Koreas neue, einzigartige Veranstaltungsorte',
  },
  content: {
    rendered:
      '<figure id="attachment_18559" aria-describedby="caption-attachment-18559" style="width: 618px" class="wp-caption aligncenter"><img class="wp-image-18559 size-large" src="https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-1200x801.jpg" alt="" width="618" height="413" srcset="https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-1200x801.jpg 1200w, https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-600x400.jpg 600w, https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-768x513.jpg 768w, https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-110x75.jpg 110w, https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300.jpg 1500w" sizes="(max-width: 618px) 100vw, 618px" /><figcaption id="caption-attachment-18559" class="wp-caption-text">Foto: Pairat Pinijkul / shutterstock.com</figcaption></figure>\n<p>Aufgrund eines exzellenten IKT-Umfelds und seiner kulturellen Attraktivität hat sich Korea als effizientes, innovatives Land für Unternehmen und Kongresse in der MICE-Branche etabliert. Hinsichtlich der Anzahl internationaler Veranstaltungen im Land belegte Korea in den letzten Jahren unter den Mitgliedern der Union der Internationalen Verbände (UIA) den ersten und zweiten Platz. Seit dem Ausbruch von Covid-19 hat die schnelle und gut organisierte Reaktion es dem Land ermöglicht, für die Sicherheit vieler Unternehmen und Konventionen zu sorgen. Mit zahlreichen Support-Programmen unterstützt Korea Verbände und Organisationen bei der Planung und Ausrichtung von Events und baut nach wie vor das Angebot an vielfältigen Eventlocations im Land aus.</p>\n',
    protected: false,
  },
  excerpt: {
    rendered:
      '<p>Aufgrund eines exzellenten IKT-Umfelds und seiner kulturellen Attraktivität hat sich Korea als effizientes, innovatives Land für Unternehmen und Kongresse in der MICE-Branche etabliert.</p>\n',
    protected: false,
  },
  featured_media: 18559,
  categories: [5, 1, 72, 170],
  acf: {
    // sponsored_by: '100HERZ',
    should_push: false,
    featured_slider: true,
    news_slider: true,
    pinned: true,
  },
  featured_image_thumb:
    'https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-300x300.jpg',
  featured_image_medium:
    'https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-600x400.jpg',
  categories_names: ['Destinationen', 'News', 'Reportagen', 'Sponsored content'],
}
