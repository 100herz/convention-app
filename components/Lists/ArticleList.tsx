import React from 'react'
import { Dimensions, FlatList, Image, ImageStyle, StyleSheet, TextStyle, View, ViewStyle } from 'react-native'
import HTML from 'react-native-render-html'
import { useNavigation } from '@react-navigation/native'
import { StackNavigationProp } from '@react-navigation/stack'
import { Ionicons } from '@expo/vector-icons'

import Text from '@components/UI/Text'
import Button from '@components/UI/Button'
import { RootStackParamList } from '@navigations/ArticleNavigator'
import { Article } from '@models/article'
import { getLocaleLongDate } from '@utils/date-time'
import { colors, defaultStyles, DefaultStyles, fonts } from '@styles/theme'

interface Props {
  data: Article[]
}

const ArticleList: React.FC<Props> = ({ data }) => {
  const navigation = useNavigation<StackNavigationProp<RootStackParamList>>()

  const ArticleComponent = ({ item }: { item: Article }) => (
    <View style={styles.articleContainer}>
      <View style={styles.imageColumn}>
        {item._embedded['wp:featuredmedia'] ? (
          <Image style={styles.image} source={{ uri: item._embedded['wp:featuredmedia'][0].source_url }} />
        ) : (
          <View style={styles.noImage} testID="no-image">
            <Text style={styles.noImageText}>NO</Text>
            <Text style={styles.noImageText}>IMAGE</Text>
            <Text style={styles.noImageText}>SET</Text>
          </View>
        )}
      </View>
      <View style={styles.textColumn}>
        <HTML baseFontStyle={styles.title} html={`${item.title.rendered}`} />
        <View style={styles.dateContainer}>
          <Ionicons name="ios-clock" size={12} color={colors.accentColor} />
          <Text style={styles.date}>{getLocaleLongDate(new Date(item.date_gmt))}</Text>
        </View>
        <Button
          textStyle={{ fontSize: 12 }}
          title="Details"
          onPress={() => navigation.navigate('ArticleScreen', { postId: item.id })}
          testID="button"
        />
      </View>
    </View>
  )

  return (
    <FlatList<Article> keyExtractor={item => item.id.toString()} data={data} renderItem={ArticleComponent}></FlatList>
  )
}

interface Styles extends DefaultStyles {
  articleContainer: ViewStyle
  imageColumn: ViewStyle
  image: ImageStyle
  noImage: ViewStyle
  noImageText: TextStyle
  textColumn: ViewStyle
}

const styles = StyleSheet.create<Styles>({
  ...defaultStyles,
  articleContainer: {
    alignItems: 'center',
    flexDirection: 'row',
    padding: 15,
  },
  imageColumn: {
    width: 100,
    height: 100,
    justifyContent: 'center',
    borderRadius: 15,
  },
  image: {
    flex: 1,
    borderRadius: 15,
  },
  noImage: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.grayLight,
    borderRadius: 15,
  },
  noImageText: {
    color: 'black',
    fontFamily: fonts.sansBold,
  },
  textColumn: {
    width: Dimensions.get('screen').width - 130,
    paddingHorizontal: 15,
    alignItems: 'flex-start',
  },
  title: {
    fontFamily: fonts.sans,
    fontSize: 13,
    color: colors.primaryColor,
  },
})

export default ArticleList
