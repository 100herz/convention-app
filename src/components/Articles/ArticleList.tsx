import React from 'react'
import { FlatList } from 'react-native'

import ArticlePreview from '@components/Articles/ArticlePreview'
import { Article } from '@models/article'

interface Props {
  data: Article[]
}

const ArticleList: React.FC<Props> = ({ data }) => {
  return (
    <FlatList<Article>
      keyExtractor={item => item.id.toString()}
      data={data}
      renderItem={({ item }: { item: Article }) => <ArticlePreview article={item} />}
    />
  )
}

export default ArticleList
