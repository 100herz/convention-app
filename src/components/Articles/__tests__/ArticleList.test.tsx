import React from 'react'
import { render } from '@testing-library/react-native'

import ArticleList from '../ArticleList'
import { article } from '@__mocks__/article'

export const mockedNavigate = jest.fn()
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native')
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
    useNavigationParam: jest.fn(jest.requireActual('@react-navigation/native').useNavigationParam),
  }
})

describe('<ArticleList />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<ArticleList data={[article]} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
