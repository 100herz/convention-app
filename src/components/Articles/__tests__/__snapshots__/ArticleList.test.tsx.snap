// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleList /> renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "acf": Object {
          "featured_slider": false,
          "news_slider": true,
          "pinned": false,
          "should_push": false,
          "sponsored_by": "XYZ",
        },
        "categories": Array [
          6,
          1,
        ],
        "categories_names": Array [
          "Hotels",
          "News",
        ],
        "content": Object {
          "protected": false,
          "rendered": "<p>Durch smarte Raumanpassungen, große Flexibilität bei Buchungen und Stornierungen sowie die Etablierung eines extern zertifizierten und weit über den Standard hinaus gehenden Sicherheits- und Hygienekonzeptes hat sich Leonardo Hotels umfassend auf das „New Normal“ und die geänderten Anforderungen ihrer MICE-Kunden an zukunftsfähigen Veranstaltungsformaten eingestellt. Neue Pausenkonzepte, digitalen Helfer, verlässliche Partner sowie virtuelle und hybride Veranstaltungsformate umfassen die Angebotsvielfalt der Leonardo Hotels.  Diese sind überzeugt, dass persönliche Meetings mit Sicherheit wieder Zukunft haben.</p>
",
        },
        "date_gmt": "2020-11-11T08:25:41",
        "excerpt": Object {
          "protected": false,
          "rendered": "<p>Durch neue Sicherheits- und Hygienekonzeptes hat sich Leonardo Hotels umfassend auf das „New Normal“ und die geänderten Anforderungen an zukunftsfähigen Veranstaltungsformaten eingestellt.</p>
",
        },
        "featured_image_medium": "https://www.convention-net.de/wp-content/uploads/2020/11/iStock-1250329916-Personen_mit_Masken-w1500-h1300-600x400.jpg",
        "featured_image_thumb": "https://www.convention-net.de/wp-content/uploads/2020/11/iStock-1250329916-Personen_mit_Masken-w1500-h1300-300x300.jpg",
        "featured_media": 18571,
        "id": 18569,
        "title": Object {
          "rendered": "„New Normal“ bei Leonardo Hotels &#8211; mit neuen MICE-Formaten und Konzepten",
        },
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 10,
              "padding": 15,
            }
          }
          testID="sponsored-article"
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "flex": 1,
                "height": 650,
                "resizeMode": "cover",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://www.convention-net.de/wp-content/uploads/2020/11/iStock-1250329916-Personen_mit_Masken-w1500-h1300-600x400.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": 650,
                    "width": undefined,
                  },
                  Object {
                    "borderRadius": 15,
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "#333333",
                  "marginLeft": -15,
                  "marginTop": 50,
                  "maxWidth": "70%",
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                }
              }
              testID="sponsored-text-container"
            >
              <Text
                style={
                  Object {
                    "color": "#dfe3e8",
                    "fontFamily": "sans",
                    "fontSize": 14,
                    "lineHeight": 17.5,
                  }
                }
              >
                Sponsored by
                 
                <Text
                  style={
                    Object {
                      "color": "#dfe3e8",
                      "fontFamily": "sans-bold",
                      "fontSize": 15,
                      "lineHeight": 22.5,
                    }
                  }
                >
                  XYZ
                </Text>
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "padding": 15,
              }
            }
          >
            <View
              style={Object {}}
            >
              <Text
                allowFontScaling={true}
                selectable={false}
                style={
                  Array [
                    Object {},
                    Object {},
                  ]
                }
              >
                <Text
                  allowFontScaling={true}
                  style={
                    Object {
                      "fontFamily": "serif-bold",
                      "fontSize": 20,
                      "lineHeight": 25,
                    }
                  }
                >
                  „New Normal“ bei Leonardo Hotels – mit neuen MICE-Formaten und Konzepten
                </Text>
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
